### PR TITLE
combo cases: drop all but the most recent issue

### LINKED ISSUE #910: Combo cases/deaths producing duplicate geos
It seems that combo cases/deaths is pulling some geo ids from both usa-facts _and_ jhu-csse, which causes the archiver to balk. here's a sample file with duplicate geos:

```
$ for f in receiving/*.csv; do echo $f; cut -d, -f1 $f | sort | uniq -d; done 
[...]
receiving/20210227_county_confirmed_7dav_cumulative_num.csv
46000
46075
46107
46129
46135
```

Here's the error we get from archiver:

```
Traceback (most recent call last):
  File "/home/krivard/projects/covid/auto/covidcast-indicators/_delphi_utils_python/delphi_utils/archive.py", line 89, in diff_export_csv
    after_df_cmn = after_df.reindex(common_idx)
  File "/home/krivard/projects/covid/auto/covidcast-indicators/combo_cases_and_deaths/env/lib/python3.8/site-packages/pandas/util/_decorators.py", line 312, in wrapper
    return func(*args, **kwargs)
  File "/home/krivard/projects/covid/auto/covidcast-indicators/combo_cases_and_deaths/env/lib/python3.8/site-packages/pandas/core/frame.py", line 4176, in reindex
    return super().reindex(**kwargs)
  File "/home/krivard/projects/covid/auto/covidcast-indicators/combo_cases_and_deaths/env/lib/python3.8/site-packages/pandas/core/generic.py", line 4811, in reindex
    return self._reindex_axes(
  File "/home/krivard/projects/covid/auto/covidcast-indicators/combo_cases_and_deaths/env/lib/python3.8/site-packages/pandas/core/frame.py", line 4022, in _reindex_axes
    frame = frame._reindex_index(
  File "/home/krivard/projects/covid/auto/covidcast-indicators/combo_cases_and_deaths/env/lib/python3.8/site-packages/pandas/core/frame.py", line 4041, in _reindex_index
    return self._reindex_with_indexers(
  File "/home/krivard/projects/covid/auto/covidcast-indicators/combo_cases_and_deaths/env/lib/python3.8/site-packages/pandas/core/generic.py", line 4877, in _reindex_with_indexers
    new_data = new_data.reindex_indexer(
  File "/home/krivard/projects/covid/auto/covidcast-indicators/combo_cases_and_deaths/env/lib/python3.8/site-packages/pandas/core/internals/managers.py", line 1301, in reindex_indexer
    self.axes[axis]._can_reindex(indexer)
  File "/home/krivard/projects/covid/auto/covidcast-indicators/combo_cases_and_deaths/env/lib/python3.8/site-packages/pandas/core/indexes/base.py", line 3476, in _can_reindex
    raise ValueError("cannot reindex from a duplicate axis")
ValueError: cannot reindex from a duplicate axis
```

### PR DESCRIPTION
**Pull request by krivard**
### Description
If any region+date was updated more than once in the last week, then #906 accidentally included all retrieved issues. Luckily, this caused problems with the archiver, otherwise we wouldn't have known about the problem.

This PR drops duplicates based on `geo_id` and `timestamp`. It also refactors the unstable_sources test, since it used to rely on a blissful ignorance of duplicates

### Changelog
Itemize code/test/documentation changes and files added/removed.
- run.py:combine_usafacts_and_jhu() - drop duplicates, keeping the most recent issue
- test_run.py: add test for multiple issues, & refactor unstable_sources test to use mock instead of hacky lambdas

### Fixes 
- Fixes #910

### CODE REVIEW

File: combo_cases_and_deaths/tests/test_run.py [hunk 1]
```diff
@@ -77,6 +97,30 @@ def test_unstable_sources():
...
+    result = combine_usafacts_and_jhu("confirmed_incidence_num", "county", date_range=(0, 1), fetcher=mock_covidcast_signal)
+    pd.testing.assert_frame_equal(
+        result,
+        pd.DataFrame({
+            "geo_id": ["01000"],
+            "val": [10],
+            "timestamp": [20200101],
+            "issue": [20200104]
+        },
+                     index=[1])
```
Comment (chinandrew) [inline combo_cases_and_deaths/tests/test_run.py:121]: nit: wonky indent

**Review by chinandrew — APPROVED** [review]
:shipit: :shipit: :shipit:
